File: 7kyu/Strong Number/C/src/solution.c
// This is an independent project of an individual developer. Dear PVS-Studio, please check it.
// PVS-Studio Static Code Analyzer for C, C++ and C#: http://www.viva64.com
#include "solution.h"

#include <assert.h>

static int fact(int n) noexcept
{
	int res = 1;
	for (int i = 1; i <= n; ++i) {
		res *= i;
	}
	return res;
}

const char* strong_num(const int number) noexcept
{
	assert(number > 0);

	int temp = number;
	int res = 0;
	while (temp > 0 && res < number) {
		res += fact(temp % 10);
		temp /= 10;
	}
	if (res == number) {
		return "STRONG!!!!";
	}

	return "Not Strong !!";
}

File: 7kyu/Strong Number/Typescript/src/solution.ts
const myAssert = (condition: boolean) => {
    if (!condition) {
        throw new Error("assertion failed");
    }
};

const fact = (n: number): number => {
    let res = 1;
    for (let i = 1; i <= n; ++i) {
        res *= i;
    }

    return res;
};

export const strongNumber = (num: number): string => {
    myAssert(num > 0);

    let temp = num;
    let res = 0;
    while (temp > 0 && res < num) {
        res += fact(temp % 10);
        temp /= 10;
    }
    if (res === num) {
        return "STRONG!!!!";
    }

    return "Not Strong !!";
};
